daemons/execd/cts-exec-helper.c

```diff
@@ -508,132 +517,52 @@ generate_params(void)
-        flag = pcmk__next_cli_option(argc, argv, &option_index, NULL);
-        if (flag == -1)
-            break;
-
-        switch (flag) {
-            case '?':
-                pcmk__cli_help(flag, CRM_EX_OK);
-                break;
-            case 'V':
-                ++options.verbose;
-                crm_bump_log_level(argc, argv);
-                break;
-            case 'Q':
-                options.quiet = 1;
-                options.verbose = 0;
-                break;
-            case 'l':
-                options.listen = optarg;
-                break;
-            case 'w':
-                options.no_wait = TRUE;
-                break;
-            case 'R':
-                options.is_running = TRUE;
-                break;
-            case 'n':
-                options.exec_call_opts = lrmd_opt_notify_orig_only;
-                break;
-            case 'o':
-                options.exec_call_opts = lrmd_opt_notify_changes_only;
-                break;
-            case 'c':
-                options.api_call = optarg;
-                break;
-            case 'a':
-                options.action = optarg;
-                break;
-            case 'r':
-                options.rsc_id = optarg;
-                break;
-            case 'x':
-                if(optarg) {
-                    options.cancel_call_id = atoi(optarg);
-                }
-                break;
-            case 'P':
-                options.provider = optarg;
-                break;
-            case 'C':
-                options.class = optarg;
-                break;
-            case 'T':
-                options.type = optarg;
-                break;
-            case 'i':
-                if(optarg) {
-                    options.interval_ms = crm_parse_interval_spec(optarg);
-                }
-                break;
-            case 't':
-                if(optarg) {
-                    options.timeout = atoi(optarg);
-                }
-                break;
-            case 's':
-                if(optarg) {
-                    options.start_delay = atoi(optarg);
-                }
-                break;
-            case 'k':
-                key = optarg;
-                if (key && val) {
-                    options.params = lrmd_key_value_add(options.params, key, val);
-                    key = val = NULL;
-                }
-                break;
-            case 'v':
-                val = optarg;
-                if (key && val) {
-                    options.params = lrmd_key_value_add(options.params, key, val);
-                    key = val = NULL;
-                }
-                break;
-            case 'S':
-                options.use_tls = TRUE;
-                break;
-            default:
-                ++argerr;
-                break;
-        }
-    }
+    pcmk__common_args_t *args = pcmk__new_common_args(SUMMARY);
+    gchar **processed_args = pcmk__cmdline_preproc(argv, "aciklnorstvwxCPT");
```

Comment:
We don't have to check any arguments here because this tool is internal to pacemaker, so we control how it's called

Reply:
I think we could just get away with not calling pcmk__cmdline_preproc here then (and in cts-fence-helper.c for that matter), but if we do that we will also have to call g_set_prgname manually.  I'll just change the last arg to NULL here and put a comment telling my future self why I'm doing that.